core/spec/models/spree/taxon_spec.rb

```diff
@@ -27,6 +27,11 @@
         end
 
         if Spree::Config.taxon_attachment_module == Spree::Taxon::PaperclipAttachment
+          it "returns false if destroying the attachment fails" do
+            allow(taxon.icon).to receive(:destroy).and_return(false)
+            expect(taxon.destroy_attachment(:icon)).to be_falsey
```

Comment:
Just to be sure, do you think it's worth checking if the file_name is still there?

```
expect(taxon.reload.icon_file_name).to_not be_present
```

Reply:
@kennyadsl the paperclip attributes such as `icon_file_name` will still be present because the destroy failed and the attachment still exists. I don't think it's necessary to test this, because paperclip does the setting of these attributes to `nil` on destroying. If Solidus did this, I think it would be necessary.

I am happy to add it if you feel strong about it, and I am interested to hear more opinions about it.

Reply:
Sorry I meant:

```
expect(taxon.reload.icon_file_name).to be_present
```

By the way, I got your point. If you think this is already covered by Paperclip, we are good as is. 👍